File: 수학(구현)/백준_2847_게임을 만든 동준이/게임을 만든 동준이.py
n = int(input())
arr = [0 for _ in range(n)]

for i in range(n):
    arr[i] = int(input())

sum = 0
for i in range(n-1, 0, -1):
    if arr[i-1] >= arr[i]:
        sum += (arr[i-1] - (arr[i] - 1))
        arr[i-1] = arr[i] - 1

print(sum)

File: 수학(구현)/백준_2847_게임을 만든 동준이/게임을 만든 동준이.cpp
#include <iostream>
#include <string.h>
using namespace std;


int main() {
    int n = 0;
    scanf("%d", &n);
    int arr[100] = {0};

    for(int i = 0; i< n; i++) scanf("%d", &arr[i]);

    int sum = 0;
    for(int i = n - 1; i > 0; i--) {
        if(arr[i-1] >= arr[i]){
            sum += (arr[i-1] - (arr[i] - 1));
            arr[i-1] = arr[i] - 1;
        }
    }

    printf("%d", sum);
}

